### PR TITLE
feat: dockerised Frontend-22 for prod

### PR DESCRIPTION
**Pull request by ZaBarudo**
@Hedhav Review please.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM node:19-alpine as base
+
+RUN mkdir -p /app
+COPY package.json /app
+COPY yarn.lock /app
+WORKDIR /app
```
Comment (Hedhav) [inline Dockerfile:6]: WORKDIR is typically specified right under the `FROM <image_name>` line

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+COPY . .
+RUN yarn run build
+
+
+FROM nginx:1.23.3-alpine as prod
+
+COPY --from=base /app/dist /usr/share/nginx/html
+COPY nginx.conf /etc/nginx/conf.d/default.conf
+EXPOSE 80
+CMD ["nginx", "-g", "daemon off;"]
```
Comment (Hedhav) [inline Dockerfile:17]: New line at EOF

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+version: '3.9'
+
+services:
+  server:
+    image: fest_frontend:prod
```
Comment (Hedhav) [inline docker-compose.yaml:5]:
```suggestion
    image: main_site_22:prod
```

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    build:
+      context: .
+      target: prod
+    container_name: fest_frontend
```
Comment (Hedhav) [inline docker-compose.yaml:9]:
```suggestion
    container_name: main_site_22
```

**Review by Hedhav — CHANGES_REQUESTED** [review]
Add new lines at EOF

Space out the Dockerfile better

**Comment by ZaBarudo** [comment]
@Hedhav Done making changes

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
+FROM node:19-alpine as base
+
+WORKDIR /app
+RUN mkdir -p .
+COPY package.json .
+COPY yarn.lock .
```
Comment (Hedhav) [inline Dockerfile:6]:
```suggestion
COPY package.json package.json
COPY yarn.lock yarn.lock
```

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+services:
+  server:
+    image: main_site_22:prod
+    build:
+      context: .
+      target: prod
+    container_name: main_site_22
+    ports:
+      - 3000:80
+    restart: always
```
Comment (Hedhav) [inline docker-compose.yaml:12]: Order:
    container_name
    image
    build: 
    ports:
    restart:

**Review by Hedhav — CHANGES_REQUESTED** [review]
New lines in EOF please

**Review by Hedhav — APPROVED** [review]

**Review by Ram-20062003 — APPROVED** [review]

**Pull request merged**